src/Take.Elephant.Sql/DbConnectionExtensions.cs

```diff
@@ -88,7 +90,7 @@ public static class DbConnectionExtensions
                     tableName = databaseDriver.ParseIdentifier(tableName),
                     filter = SqlHelper.GetAndEqualsStatement(databaseDriver, filterValues)
                 },
-                filterValues.Select(k => k.ToDbParameter(databaseDriver)));
+                filterValues.Select(k => k.ToDbParameter(databaseDriver, columnTypes)));
```

Comment:
I think it worth to introduce a higher level extension method to better encapsulate the parameter generation, moving the projection (Select call) to it.

The signature could be (already adopting the suggestion for receiving the ITable instead of some parts of it; but you could use the `columnTypes` Dictionary as well):
```
public static IEnumerable<DbParameter> ToDbParameters(this IDictionary<string, object> parameters, IDatabaseDriver databaseDriver, ITable table)
````

So this call could be written as:
```
filterValues.ToDbParameters(databaseDriver, columnTypes);